subsys/bluetooth/audio/tbs_client.c

```diff
@@ -600,6 +599,69 @@ static uint8_t notify_handler(struct bt_conn *conn,
 	return BT_GATT_ITER_CONTINUE;
 }
 
+/* Common function to read tbs_client strings which may require long reads */
+static uint8_t handle_string_long_read(struct bt_conn *conn, uint8_t err,
+				       struct bt_gatt_read_params *params, const void *data,
+				       uint16_t length, bt_tbs_client_read_string_cb cb)
+{
+	struct bt_tbs_instance *inst = CONTAINER_OF(params, struct bt_tbs_instance, read_params);
+	uint16_t offset = params->single.offset;
+	uint8_t inst_index = tbs_index(conn, inst);
+	const char *received_string;
+	int tbs_err = err;
+
+	if ((tbs_err == 0) && (data != NULL) &&
+	    (net_buf_simple_tailroom(&inst->net_buf) < length)) {
+		tbs_err = BT_ATT_ERR_INSUFFICIENT_RESOURCES;
+	}
+
+	if (tbs_err != 0) {
+		LOG_DBG("err: %d", tbs_err);
+		if (cb != NULL) {
+			cb(conn, tbs_err, inst_index, NULL);
+		}
+
+		(void)memset(params, 0, sizeof(*params));
+
+		return BT_GATT_ITER_STOP;
+	}
+
+	if (data != NULL) {
+		/* Get data and try to read more using read long procedure */
+		LOG_DBG("Read (offset %u): %s", offset, bt_hex(data, length));
+
+		net_buf_simple_add_mem(&inst->net_buf, data, length);
+
+		return BT_GATT_ITER_CONTINUE;
+	}
+
+	if (inst->net_buf.len == 0) {
+		received_string = NULL;
+	} else {
+		char *str_data;
+		uint16_t str_length = MIN(inst->net_buf.len, CONFIG_BT_TBS_MAX_URI_LENGTH);
```

Comment:
Why do we need the `MIN` here? 

And if so, should the `MIN` not use the net_buf size, rather than `CONFIG_BT_TBS_MAX_URI_LENGTH`?

Reply:
Good question :-) 
I will just set it to the length of the buffer.